### ISSUE TITLE
Quotes being used breaking alt text

### ISSUE THREAD

**Issue by srancour** [department-of-veterans-affairs/va.gov-cms#17444]
## Problem Statement
Editors who are using quotes in their alt text are breaking the alt tag on <img> html elements.

## Description or Additional Context 
Editors are using quotes especially when setting alt text for images with words in them by saying things like in a [recent sentry notification](https://dsva.slack.com/archives/CJT90C0UT/p1709761701286139)

> with the text that reads "national nutrition month" in the middle

In implementation this makes the alt tag stop at the word before the quote and nothing else the editor wanted is included.

## Steps for Implementation
- [x] Technical review of the solution to either remove the ability to put quotation marks in alt text or put in the ability to escape quotation marks to determine what is technically feasible, knowing the skillset of the CMS team.
- [x] Review of the alt text data to understand how quotations are currently being use to help inform the solution and if there is any change management needs to happen. 
- [x] Review the alt text data to determine the scope and volume of this problem. 

## Acceptance Criteria
- [x] A plan of action on how to solve this problem, both previously added alt text and future alt text. 
- [x] An understanding of how to prioritize this fix. 
- [x] Requires accessibility review

## Team
Please check the team(s) that will do this work.

- [x] CMS Team

**Comment by ian-sears** [comment]
A Sample page for testing:
https://prod.cms.va.gov/salisbury-health-care/about-us/leadership

**Comment by srancour** [comment]
Pulled down the CSV of the current images in prod and there are 692 instances of `"` being used in the alt-text column. There are 261 images currently in the system with `"` somewhere in the alt-text.

**Comment by srancour** [comment]
Not sure how, but some editors have also used `“` instead of `"` which is on 55 images. In excel the `“` shows as `‚Äú`. This style quote does not break the alt-text field.

**Comment by srancour** [comment]
Talking with @laflannery there is guidance for using curly quotes instead of straight ones in the [VA Design System](https://design.va.gov/content-style-guide/punctuation#quotation-marks), which it does look like what those 55 images are using and why it's not breaking the alt-text. There is also a [ticket from the previous team](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5885) to do this work that was never completed. This probably would also solve the issue @ian-sears also saw with the staff profiles. @laflannery brought up that this is also all throughout the CMS with facility names and operating statuses as other examples of where we're having issues on the front end.

Replacing editor's straight double quotes in the CMS with curly double quotes could pose search issues, which we would have to account for if we went this direction. @EWashb [brought up that issue](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5885#issuecomment-1126239597) on the ticket when it was with the last team.

**Comment by aklausmeier** [comment]
[Content style guide](https://design.va.gov/content-style-guide/punctuation#quotation-marks) related to quotation marks:

> Use curly quotation marks (ex. “quotation marks”) instead of straight up and down ones

**Comment by MDomngz** [comment]
My question is: Why would an editor choose to use a different type of "quote" mark? Is that intentional or accidental based on their keyboard settings?

**Comment by srancour** [comment]
By default keyboards use straight up and down quotes. The way some of the curly quotes that are currently in Drupal probably got there is from editors copy/pasting from another source that was using curly quotes. [Copy/pasting for the alt-text is something I have found and documented](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17452#issuecomment-1989006424) coming from the data for the alt-text.

**Comment by srancour** [comment]
## Tldr 

### Suggestions:

Talking to @edmund-dunn, we believe the best solution for this, right now, should be that we focus on the Drupal end. This would be a two step process combining the Alt-text field solution and the Fix current alt-text solutions below. We will need 2 tickets created for this work:

1. Ticket to update the alt-text field where, on submit, it does a regex check for double straight quotes and silently replaces them with double curly quotes for the editor.
2. Ticket to create a script that will run through the current alt-text in the using regex to check for double straight quotes and replace them with double curly quotes.

### Future considerations:

Once the future of AP is known, the Build solution below may be viable to go for.

## Longer discussion

### Build solution:

We discussed processing double quotes on the build side where it would change double straight quotes into double curly quotes while building the front end. This has a lot of unknowns that would make this a major lift.

#### Pros:

- Probably the correct way to fix this long term
- Should also solve the issue #18193
- Should solve any other issues that double straight quotes might be causing that we haven't found yet
- Should mitigate any new work that is done on the Drupal side that might break part of the front end with editors adding double straight quotes.
- Should make sure the front end is showing the curly quotes, per the style guide, from everything that comes out of Drupal
- Editors won't have any changes on them

#### Cons:

- Affected by the uncertainty of the future of AP
    - If AP continues, we could add on a helper tool that would do this work which could be relatively easy to implement
    - If AP doesn't continue, there would be a major lift to get into the current build process to figure out where this could even be implemented.
- Uncertainty of size of lift

#### Suggestions:

We wouldn't suggest this solution at this time. This solution, while may be the correct way for a long term solution for the entirety of Drupal, would be a major lift and size would be determined by the future of AP.

### Alt-text field solution:

We discussed processing double quotes on the alt-text field where it would change double straight quotes into double curly quotes on submission from the editor. This work has little to no unknowns and would be a minor lift.

#### Pros:

- Very minor lift
    - Familiarity with the alt-text field
    - Simple regex statement
- Sitewide team could use this solution as reference for fixing part of #18193 

#### Cons:

- Would only fix the alt-text field
- Editors who search for the alt-text in the images with quotes won't be able to search with that method for images they submit after this change
    - Would require a second lift on the search to be able to have curly quotes come up with a search with straight quotes
- Would need a second lift of the Fix current alt-text solution below to fix the current alt-text in the system
- Sitewide would have do their own separate lift to solve #18193

#### Suggestions:

We would suggest this solution along with the Fix current alt-text Solution below. This solution, while may not fix everything in Drupal for double straight quote usage, is a small lift, viable, and easy way to get the alt-text added from editors fixed quickly.

### Fix current alt-text solution:

We discussed a script with a simple regex that would run through all of the alt-text in the system to replace double straight quotes with double curly quotes.

#### Pros:

- Very minor lift
    - Work from the PDF removal script can be directly applied
    - Simple regex statement
- Sitewide team could use this solution as a reference for fixing part of #18193

#### Cons:

- Would only fix the current alt-text in the system
- Editors who search for the alt-text in the images with quotes won't be able to search with that method for images currently in the system
    - Would require a second lift on the search to be able to have curly quotes come up with a search with straight quotes
- Would need a second lift of the Alt-text field solution above to fix the alt-text input by editors
- Sitewide would have do their own separate lift to solve #18193

#### Suggestions:

We would suggest this solution along with the Alt-text Field Solution above. This solution, while may not fix everything in Drupal for double straight quote usage, is a small lift, viable, and easy way to get the alt-text currently in the system fixed quickly.

### Drupal-wide input solution

We discussed processing double quotes in every field in Drupal where it would change double straight quotes into double curly quotes on submission from the editor. This would be similar to the Alt-text field solution above, but Drupal-wide. This work has a lot of unknowns and would be a very major lift.

#### Pros:

- Should fix this issue for editor input Drupal-wide
- Could be made to be future proof
    - Lots of unknowns with this
- Should fix part of #18193 with no extra work for the Sitewide team

#### Cons:

- Major lift
- Lots of Unknowns
- Editors who search for quotes in anything won't be able to search for anything they submitted after this change
    - Would require a second lift on the search to be able to have curly quotes come up with a search with straight quotes
- Would need a second lift of the Fix all editor submitted data Drupal-wide solution below for everything currently in the system

#### Suggestions:

We wouldn't suggest this solution because, while it would fix the issue Drupal-wide for editor inputing new data and could be made to be future proof, this has so many unknowns from the number of different Drupal inputs and could easily be broken future-wise if another team creates a new input that can't use this solution. 

### Fix all editor submitted data Drupal-wide solution

We discussed a script with a simple regex that would run through all of the editor submitted data in the system to replace double straight quotes with double curly quotes. This would be similar to the Alt-text field solution above, but Drupal-wide. This work has a lot of unknowns and would be a very major lift.

#### Pros:

- Should fix this issue for all editor submitted data currently in the system
- Should fix part of #18193 with no extra work for the Sitewide team

#### Cons:

- Major lift
- Lots of Unknowns
    - Complicated single regex script
    - Multiple simple regex scripts
- Editors who search for quotes in anything won't be able to search for anything they submitted after this change
    - Would require a second lift on the search to be able to have curly quotes come up with a search with straight quotes
- Would need a second lift of the Drupal-wide input solution above to fix the inputs Drupal-wide

#### Suggestions:

We wouldn't suggest this solution because, while it would fix the issue Drupal-wide for currently existing editor submitted data, this has so many unknowns from the different types and locations of the data submitted by editors and how to either run a single script or run the same script over and over through every type of data.

**Comment by srancour** [comment]
Also, to make the comment here a bit easier to read and to be able to have links to the different sections, I put the information into a [Confluence page](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/3450994691/Double+Straight+Quotes+Breaking+Discussion)

**Comment by gracekretschmer-metrostar** [comment]
Next steps: do technical refinement to determine the best technical approach. Idea: can we put in a filter on the data to catch straight quotes before it gets passed to Content Build and Next Build?
